### PR TITLE
docs: adding paymaster tutorial

### PR DESCRIPTION
**Pull request by excaliborr**
closes SPO-31

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,21 +1,10 @@
-MIT License
+The MIT License (MIT)
+Copyright © 2024 Wonderland
```
Comment (0xGorilla) [inline LICENSE:2]:
```suggestion
Copyright © 2024 Wonder Ltd
```

File: README.md [hunk 2]
```diff
@@ -108,8 +109,165 @@ await testToken.mint(amountToMint);
 
 ## Adding a new paymaster to the plugin
```
Comment (0xGorilla) [inline README.md:110]: Let's move all this to another .md file called `contributors.md` or something like that, just not to fill the readme with information 99% of the users don't need

In the readme.md we should mention that contributors should go read that file, and link it.

Wdyt?

File: README.md [hunk 3]
```diff
@@ -108,8 +109,165 @@ await testToken.mint(amountToMint);
...
 
-TODO when this functionallity is finalized
+### Adding the paymaster
+
+If you operate a paymaster and would like to add support for your paymaster into the plugin please follow the steps below
+
+1. located inside `src/paymasters` you will see all other paymasters please create a typescript file for this example we will call it `ExamplePaymaster.ts`
```
Comment (0xGorilla) [inline README.md:116]:
```suggestion
1. Located inside `src/paymasters` you will see all other paymasters please create a typescript file for this example we will call it `ExamplePaymaster.ts`
```

File: README.md [hunk 4]
```diff
@@ -108,8 +109,165 @@ await testToken.mint(amountToMint);
 
 ## Adding a new paymaster to the plugin
 
-TODO when this functionallity is finalized
+### Adding the paymaster
```
[the remark this answers is not in the document]
Comment (excaliborr) [reply README.md:112]: We do at the end of the How to test your new paymaster section

File: CONTRIBUTORS.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+<br>
+
+```
+GOERLI_RPC_URL=https://example_rpc
+PRIVATE_KEY=YOUR_PK
+```
+9. Now we can test everything by running `npx hardhat run scripts/deploy.ts`
+
+10. Assuming all goes well and you are done testing your implementation feel free to make a PR to the official plugin and we will do our best to review it in a timely manner!
```
[the remark this answers is not in the document]
Comment (excaliborr) [reply CONTRIBUTORS.md:163]: Reviewer will have to test it manually as we dont have the URL before they make the PR to make calls to, I will add a note to add details on how a reviewer can test it in the comments of the users PR

**Review by 0xOneTony — APPROVED** [review]

File: LICENSE (hunk 1 above)
Comment (0xGorilla) [reply LICENSE]: we shouldn't...

**Review by 0xGorilla — APPROVED** [review]

**Pull request merged**